src/Raven.Server/Commercial/LicenseManager.cs

```diff
@@ -112,6 +112,8 @@ public LicenseManager(ServerStore serverStore)
             _serverStore = serverStore;
             _licenseHelper = new LicenseHelper(serverStore);
             _skipLeasingErrorsLogging = serverStore.Configuration.Licensing.SkipLeasingErrorsLogging;
+
+            ClusterCommandsVersionManager.AfterClusterVersionSetting += PutMyNodeInfoAfterClusterVersionSetting;
```

Comment:
now you can get this event before you initialize the storage? you should probably do it in Initialize, but remember to unsubscribe in Dispose as well

Reply:
In the context of what I've written above, I believe we should subscribe only if the cluster is not ready to execute commands yet.